hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -744,6 +744,25 @@ protected InternalCompletableFuture<Data> removeAsyncInternal(Object key) {
         }
     }
 
+    protected InternalCompletableFuture<Data> deleteAsyncInternal(Object key) {
+        Data keyData = toDataWithStrategy(key);
+        int partitionId = partitionService.getPartitionId(keyData);
+        MapOperation operation = operationProvider.createDeleteOperation(name, keyData, false);
+        operation.setThreadId(getThreadId());
+        try {
+            long startTimeNanos = Timer.nanos();
+            InvocationFuture<Data> future = operationService.invokeOnPartition(SERVICE_NAME, operation, partitionId);
```

Comment:
The remove variant uses `invokeOnPartitionAsync`, can you also do the same here?